ogm/occupancy_grid_map.py

```diff
@@ -96,13 +97,20 @@ def recenter_initial_positions(self, module_positions, final_module_positions, g
             pos[2] + offset[2]
         )
     
+    # Calculate the "final" offset to move module 1 to grid center (in the case that module 1's final position deviates from its initial position)
```

Comment:
Rest looks fine. 
Does the function (check_final works correctly), which compares the current position with final position works fine? 

There we are comparing module current position with final position, but both recentred with different modules.